app/controllers/session_controller.rb

```diff
@@ -634,6 +622,12 @@ def check_local_login_allowed(user: nil, check_login_via_email: false)
 
   private
 
+  def connect_verbose_warn(&blk)
```

Comment:
Why a block? Any advantages over just passing **args?

Reply:
It just means the string doesn't get computed when it's not needed